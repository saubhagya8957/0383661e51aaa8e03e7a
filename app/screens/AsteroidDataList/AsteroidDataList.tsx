import React, { Component } from "react";
import { View, SafeAreaView, Text, TextInput, TouchableOpacity, Alert } from 'react-native';
import styles from './AsteroidDataListStyle';


interface Props {
    navigation: any;
    id: string;
    route: any;
}

interface S {
    asteroidData: any;
    asteroidDataList: any;
}

interface SS {
    id: any
}

class AsteroidDataList extends Component<Props, S, SS> {
    constructor(props: Props) {
        super(props);
        this.state = {
            asteroidData: this.props.route.params.asteroidData,
            asteroidDataList: [],
        }
    }

    componentDidMount = () => {
        console.log('@@@ Asteroid Params Response ===========', this.props.route.params.asteroidData);
    }

    componentWillUnmount = () => {

    }

    render() {
        const { name, nasa_jpl_url, is_potentially_hazardous_asteroid, id } = this.state.asteroidData;
        return (
            <SafeAreaView style={styles.container}>
                <View style={styles.asteroidDataCell}>
                    <Text style={styles.key}>Name: {name}</Text>
                    <Text style={styles.key}>NASA JPL URL: {nasa_jpl_url}</Text>
                    <Text style={styles.key}>IS POTENTIALLY HAZARDOUS ASTEROID: {String(is_potentially_hazardous_asteroid)}</Text>
                </View>
            </SafeAreaView>
        );
    }
}

export default AsteroidDataList;