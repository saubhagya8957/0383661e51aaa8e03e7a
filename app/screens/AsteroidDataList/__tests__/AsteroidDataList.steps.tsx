import React from "react";
import { defineFeature, loadFeature } from 'jest-cucumber';
import { shallow, ShallowWrapper, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AsteroidDataList from "../AsteroidDataList";

configure({ adapter : new Adapter() });

const screenProps = {
    navigation: {
        navigate: jest.fn(),
    },
    route: {
        params: {
            asteroidData: {
                name: 'Asteroid',
                nasa_jpl_url: '',
                is_potentially_hazardous_asteroid: '',
            }
        }
    },
    id: 'Asteroid Data List'
}

const feature = loadFeature('./app/screens/AsteroidDataList/__tests__/features/AsteroidDataList.scenarios.feature');

defineFeature(feature, (test) => {
    test('User navigates Asteroid Specific Data', ({ given, when, then }) => {
        let AsteroidDataListWrapper: ShallowWrapper;
        let AsteroidDataListInstance: AsteroidDataList;

        given('I am user loading Asteroid Specific Data Screen', () => {
            AsteroidDataListWrapper = shallow(<AsteroidDataList {...screenProps} />);
        });

        when('I just navigate to the Asteroid Specific Data Screen', () => {
            AsteroidDataListInstance = AsteroidDataListWrapper.instance() as AsteroidDataList;
        });

        then('Asteroid Specific Data will mount with out any errors', () => {
            expect(AsteroidDataListWrapper).toBeTruthy();
            expect(AsteroidDataListWrapper).toMatchSnapshot();
        });

        then('I can unmount the screen with out any errors', () => {
            AsteroidDataListInstance.componentWillUnmount();
            expect(AsteroidDataListWrapper).toBeTruthy();
            expect(AsteroidDataListWrapper).toMatchSnapshot();
        });
    })
})