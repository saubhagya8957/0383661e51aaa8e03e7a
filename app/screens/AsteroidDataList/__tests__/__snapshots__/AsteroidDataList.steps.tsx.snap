// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asteroid Specific Data Screen User navigates Asteroid Specific Data 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ab8e00",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#628ead",
        "borderColor": "#000000",
        "paddingRight": 12,
        "paddingVertical": 10,
        "width": 350,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginLeft": 20,
          "marginVertical": 10,
          "width": 300,
        }
      }
    >
      Name: 
      Asteroid
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginLeft": 20,
          "marginVertical": 10,
          "width": 300,
        }
      }
    >
      NASA JPL URL: 
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginLeft": 20,
          "marginVertical": 10,
          "width": 300,
        }
      }
    >
      IS POTENTIALLY HAZARDOUS ASTEROID: 
    </Text>
  </View>
</ForwardRef(SafeAreaView)>
`;

exports[`Asteroid Specific Data Screen User navigates Asteroid Specific Data 2`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ab8e00",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#628ead",
        "borderColor": "#000000",
        "paddingRight": 12,
        "paddingVertical": 10,
        "width": 350,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginLeft": 20,
          "marginVertical": 10,
          "width": 300,
        }
      }
    >
      Name: 
      Asteroid
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginLeft": 20,
          "marginVertical": 10,
          "width": 300,
        }
      }
    >
      NASA JPL URL: 
    </Text>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginLeft": 20,
          "marginVertical": 10,
          "width": 300,
        }
      }
    >
      IS POTENTIALLY HAZARDOUS ASTEROID: 
    </Text>
  </View>
</ForwardRef(SafeAreaView)>
`;
